### PR TITLE
Mock GPU accelerator connector tests

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of #9939

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @justusschock @kaushikb11 @awaelchli @akihironitta

### CODE REVIEW

File: tests/accelerators/test_accelerator_connector.py [hunk 1]
```diff
@@ -174,13 +179,15 @@ def on_fit_start(self, trainer, pl_module):
             raise SystemExit()
 
     model = BoringModel()
-    trainer = Trainer(fast_dev_run=True, accelerator="ddp", gpus=2, callbacks=[CB()])
+    with pytest.deprecated_call(match=r"accelerator='ddp'\)` has been deprecated in v1.5"):
```
Comment (awaelchli) [inline tests/accelerators/test_accelerator_connector.py:182]: why are we adding these deprecation tests in some places and update to the new api in other places? 
Shouldn't we set strategy=ddp here?

File: tests/accelerators/test_accelerator_connector.py [hunk 2]
```diff
@@ -326,26 +340,25 @@ def on_fit_start(self, trainer, pl_module):
 @RunIf(special=True)
 def test_accelerator_choice_ddp_cpu_and_plugin(tmpdir):
     """Test that accelerator="ddp_cpu" can work together with an instance of DDPPlugin."""
-    _test_accelerator_choice_ddp_cpu_and_plugin(tmpdir, ddp_plugin_class=DDPPlugin)
+    _test_accelerator_choice_ddp_cpu_and_plugin(tmpdir, ddp_strategy_class=DDPPlugin)
```
Comment (kaushikb11) [inline tests/accelerators/test_accelerator_connector.py:343]:
```suggestion
    _test_accelerator_choice_ddp_cpu_and_strategy(tmpdir, ddp_strategy_class=DDPPlugin)
```
For other changes as well

File: tests/accelerators/test_accelerator_connector.py [hunk 3]
```diff
@@ -111,13 +112,13 @@ def on_fit_start(self, trainer, pl_module):
             raise SystemExit()
 
     model = BoringModel()
-    trainer = Trainer(fast_dev_run=True, accelerator="ddp", gpus=2, callbacks=[CB()])
+    with pytest.deprecated_call(match=r"accelerator='ddp'\)` has been deprecated in v1.5"):
+        trainer = Trainer(fast_dev_run=True, accelerator="ddp", gpus=2, callbacks=[CB()])
```
Comment (tchaton) [inline tests/accelerators/test_accelerator_connector.py:116]: Why not use strategy directly ?

File: tests/accelerators/test_accelerator_connector.py (hunk 1 above)
Comment (carmocca) [reply tests/accelerators/test_accelerator_connector.py:182]: if I'm not mistaken, @kaushikb11 duplicated most of the `accelerator` flag tests but using the `strategy` flag.

This is one of them.

**Review by tchaton — APPROVED** [review]
LGMT !

File: tests/accelerators/test_accelerator_connector.py (hunk 3 above)
Comment (carmocca) [reply tests/accelerators/test_accelerator_connector.py:116]: https://github.com/PyTorchLightning/pytorch-lightning/pull/10554#discussion_r750342837

File: tests/accelerators/test_accelerator_connector.py (hunk 2 above)
Comment (carmocca) [reply tests/accelerators/test_accelerator_connector.py]: Done

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
